src/main/java/com/es/phoneshop/dao/AbstractDao.java

```diff
@@ -1,15 +1,16 @@
 package com.es.phoneshop.dao;
 
+import com.es.phoneshop.exceptions.BaseModelNotFoundException;
 import com.es.phoneshop.exceptions.OrderNotFoundException;
-import com.es.phoneshop.model.BaseModelEntity;
+import com.es.phoneshop.model.BaseModel;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
-public abstract class AbstractDao<T extends BaseModelEntity> {
+public abstract class AbstractDao<T extends BaseModel, E extends BaseModelNotFoundException> {
```

Comment:
You don't use parmetrized attribute **E extends BaseModelNotFoundException**.
Please remove it